docs/config.rst

```diff
@@ -63,3 +63,39 @@ Creating Your Own Config
 When creating your own config, you don't need to specify any of these values,
 a config can contain any values you want. In that case however, your
 template must only use the values that your config has.
+
+Defining ``project.py``
+-----------------------
+
+You can define a file called ``project.py`` in root of your project dir. The
+python code will be executed by distgen. It is very convenient way of
+providing dynamic values into your templates.
+
+::
+
+    import subprocess
+    from distgen.project import AbstractProject
+
+    class Project(AbstractProject):
+        """ This class has to be named "Project" """
+
+        def inst_init(self, spec, template, sysconf):
+            """ performed before instantiation """
+            # let's add current date to config
+            sysconf['current_date'] = subprocess \
+                .check_output(["date"]) \
+                .decode("utf-8")
+
+        def inst_finish(self, spec, tpl, config, data):
```

Comment:
Nit, the `inst_init` and `inst_finish` arguments have different spelling -- it might look that `config` means something different from `sysconf` (the same for `tpl` vs `template`).

The arguments should be `list_of_specfiles`, `template`, `configuration`, `spec` (finish case).  Note though that the call to inst_finish() has bug (generated_spec is passed twice).